navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -0,0 +1,260 @@
+  }
+
+  private void bind() {
+    mapView = (MapView) findViewById(R.id.mapView);
+    instructionView = (InstructionView) findViewById(R.id.instructionView);
+  }
+
+  private void initMap(Bundle savedInstanceState) {
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  private void initNavigation() {
+    navigation = new MapboxNavigation(this);
+    navigation.addProgressChangeListener(this);
+
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  private void initLocation() {
+    //    locationEngine = new LocationSource(this);
+    //    locationEngine.setPriority(HIGH_ACCURACY);
+    //    locationEngine.setInterval(0);
+    //    locationEngine.setFastestInterval(1000);
+    //    locationEngine.addLocationEngineListener(this);
+    //    locationEngine.activate();
+    //
+    //    if (locationEngine.getLastLocation() != null) {
+    //      onLocationChanged(locationEngine.getLastLocation());
+    //    }
+    onLocationChanged(null);
+  }
+
+  private void initRoute() {
+    mapRoute = new NavigationMapRoute(mapView, map);
+  }
+
+  private void initCamera() {
+    camera = new NavigationCamera(this, map, navigation);
+  }
+
+  private void initLocationLayer() {
+    locationLayer = new LocationLayerPlugin(mapView, map, null);
+  }
+
+  private void unpackBundle(Intent intent) {
+    double originLng = intent.getDoubleExtra(NavigationConstants.NAVIGATION_VIEW_ORIGIN_LNG_KEY, 0);
+    double originLat = intent.getDoubleExtra(NavigationConstants.NAVIGATION_VIEW_ORIGIN_LAT_KEY, 0);
+    double destinationLng = intent.getDoubleExtra(NavigationConstants.NAVIGATION_VIEW_DESTINATION_LNG_KEY, 0);
+    double destinationLat = intent.getDoubleExtra(NavigationConstants.NAVIGATION_VIEW_DESTINATION_LAT_KEY, 0);
+    Position origin = Position.fromLngLat(originLng, originLat);
+    destination = Position.fromLngLat(destinationLng, destinationLat);
+    fetchRoute(origin, destination);
+  }
+
+  private boolean validRouteResponse(Response<DirectionsResponse> response) {
+    return response.body() != null
+      && response.body().getRoutes() != null
+      && response.body().getRoutes().size() > 0;
+  }
+
+  private void fetchRoute(Position origin, Position destination) {
+    NavigationRoute.Builder routeBuilder = NavigationRoute.builder()
+      .accessToken(Mapbox.getAccessToken())
+      .origin(origin)
+      .destination(destination);
+
+    if (locationHasBearing()) {
+      fetchRouteWithBearing(routeBuilder);
+    } else {
+      routeBuilder.build().getRoute(this);
+    }
+  }
+
+  private boolean locationHasBearing() {
+    return location != null && location.hasBearing();
+  }
+
+  private void fetchRouteWithBearing(NavigationRoute.Builder routeBuilder) {
+    routeBuilder.addBearing(location.getBearing(), 90);
+    routeBuilder.build().getRoute(this);
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  private void startNavigation(DirectionsRoute route) {
+    activateMockLocationEngine(route);
```

Comment:
Why not make this an option for the user?

Reply:
See above ^